dfastmi/io/RiverParameterData.py

```diff
@@ -136,13 +139,31 @@ def collect_int_values1(
             all_values.append(values_per_branch)
 
         return all_values
-
-
-    def collect_values_logical(
+    
+    def __read_value(self, key, branch, reach):
+        try:
+            general_value = self._config["General"][key]
+        except:
+            general_value = ''
+        
+        try:
+            branch_val = self._config[branch.name][key]
+        except:
+            branch_val = general_value
+        
+        try:
+            val = self._config[branch.name][key + str(reach.config_key_index)]
+        except:
+            val = branch_val
+        return val
+    
+    def read_key_bool(
         self,
         key: str,
+        branch: Branch,
+        reach: Reach,            
         default: Optional[bool] = None,
-    ) -> List[List[bool]]:
+    ) -> bool:
```

Comment:
please also update the documentation

Reply:
I will but unclear to me if this refactoring/change should be done in this issue or is inscope of`[DFAST-112](https://issuetracker.deltares.nl/browse/DFAST-112) Refactor RiverParameterData and related tests

Reply:
Discussed, we will leave these comments for a later refactoring.
This will be tackled in https://issuetracker.deltares.nl/browse/DFAST-112